Source/ProgressionSystemRuntime/Private/Data/PSWorldSubsystem.cpp

```diff
@@ -255,87 +262,53 @@ void UPSWorldSubsystem::AddProgressionStarActors()
 void UPSWorldSubsystem::OnTakeActorsFromPoolCompleted(const TArray<FPoolObjectData>& CreatedObjects)
 {
 	const FPSRowData& CurrentRowData = GetCurrentRow();
+	float AmountUnlocked = CurrentRowData.CurrentLevelProgression;;
 
-	float CurrentAmountOfUnlocked = 0;
-	float CurrentAmountOfLocked = 0;
-
-	//set updated amount of stars
-	if (CurrentRowData.CurrentLevelProgression >= CurrentRowData.PointsToUnlock)
-	{
-		// set required points (stars)  to achieve for a level  
-		CurrentAmountOfUnlocked = CurrentRowData.PointsToUnlock;
-		CurrentAmountOfLocked = 0;
-	}
-	else
-	{
-		// Calculate the unlocked against locked points (stars) 
-		CurrentAmountOfUnlocked = CurrentRowData.CurrentLevelProgression;
-		CurrentAmountOfLocked = CurrentRowData.PointsToUnlock - CurrentRowData.CurrentLevelProgression;
-	}
-
-	float integerPart;
-	float fractionalPart;
-	// Setup spawned widget
 	for (const FPoolObjectData& CreatedObject : CreatedObjects)
 	{
-		if (CurrentAmountOfUnlocked > 0)
+		AActor& SpawnedActor = CreatedObject.GetChecked<AActor>();
+		UStaticMeshComponent* MeshComponent = SpawnedActor.FindComponentByClass<UStaticMeshComponent>();
+
+		SpawnedStarActorsInternal.Add(&SpawnedActor);
```

Comment:
You don't need `SpawnedStarActorsInternal` array property, as you already have the same `PoolActorHandlersInternal`.

Please remove `SpawnedStarActorsInternal` and just obtain actors from Handles.

Why? Handles is hash instead of strong pointer, so your system will not block destroying or unloading of this object. Also, it's easier to maintain just one array instead of both of them

Before:
```		SpawnedActor.SetActorLocation(SpawnedStarActorsInternal[SpawnedStarActorsInternal.Num() - 2]->GetActorLocation() + CurrentRowData.OffsetBetweenStarActors);
```

After:
```
			const FPoolObjectHandle& PreviousHandle = PoolActorHandlersInternal[PoolActorHandlersInternal.Num() - 2];
			const AActor& PreviousActor = UPoolManagerSubsystem::Get().FindPoolObjectByHandle(PreviousHandle).GetChecked<AActor>();
			SpawnedActor.SetActorLocation(PreviousActor.GetActorLocation() + CurrentRowData.OffsetBetweenStarActors);
```

Reply:
I need to skip the first element since it's driving the initial star locations, so I've modified it to store the previous actor location and skip the first element if it's empty as part of https://github.com/h4rdmol/ProgressionSystem/pull/17/commits/806bb0e19562f12faa02591f761cadf9915a0c46

Reply:
@h4rdmol, `SpawnedStarActorsInternal` is still present in .h, please remove it as it is supposed to be unused now

Reply:
done as part of: https://github.com/h4rdmol/ProgressionSystem/pull/17/commits/28651578e0ad8ae2fc96694a038e6e235e77083c